pkg/network/ovn_kubernetes.go

```diff
@@ -1544,6 +1549,16 @@ func bootstrapOVN(conf *operv1.Network, kubeClient cnoclient.Client, infraStatus
 		ipsecStatus.Version = ipsecDaemonSet.GetAnnotations()["release.openshift.io/version"]
 	}
 
+	// If we are upgrading from 4.13 -> 4.14 set new API for IP Forwarding mode to Global.
+	// This is to ensure backwards compatibility.
+	if masterStatus != nil {
+		klog.Infof("4.13 -> 4.14 upgrade detected. Will set IP Forwarding API to Global mode for backwards compatibility")
+		if conf.Spec.DefaultNetwork.OVNKubernetesConfig.GatewayConfig == nil {
+			conf.Spec.DefaultNetwork.OVNKubernetesConfig.GatewayConfig = &operv1.GatewayConfig{}
+		}
+		conf.Spec.DefaultNetwork.OVNKubernetesConfig.GatewayConfig.IPForwarding = operv1.IPForwardingGlobal
```

Comment:
IIUC, we are basically keeping the IPForwarding to be global for all older clusters?
perhaps I am missing something, but are we really patching the api object to reflect this?
conf here is meant to be read only right? or is CNO actually updating this after you set this?

Reply:
maybe https://github.com/openshift/cluster-network-operator/blob/e125ef95ef43805571c8f2b275a1a1c982932d74/pkg/network/ovn_kubernetes.go#L793 is a better place for this? feels like this render will be called always versus the bootstrap logic is a one time thing?

Reply:
Yeah the agreement we came up with @knobunc and @cgoncalves was that for clusters upgrading we would enable global forwarding. This would preserve their existing functionality in case they rely on metal lb or some other feature that requires forwarding. For fresh 4.14 installs we will keep the default as disabled.

re: "is a better place for this? feels like this render will be called always versus the bootstrap logic is a one time thing?" and "or is CNO actually updating this after you set this?"

It works because after the bootstrap we do:

```
	// Bootstrap any resources
	bootstrapResult, err := network.Bootstrap(newOperConfig, r.client)
	if err != nil {
		log.Printf("Failed to reconcile platform networking resources: %v", err)
		r.status.SetDegraded(statusmanager.OperatorConfig, "BootstrapError",
			fmt.Sprintf("Internal error while reconciling platform networking resources: %v", err))
		return reconcile.Result{}, err
	}

	if !reflect.DeepEqual(operConfig, newOperConfig) {
		if err := r.UpdateOperConfig(ctx, newOperConfig); err != nil {
			log.Printf("Failed to update the operator configuration: %v", err)
			r.status.SetDegraded(statusmanager.OperatorConfig, "UpdateOperatorConfig",
				fmt.Sprintf("Internal error while updating operator configuration: %v", err))
			return reconcile.Result{}, err
		}
	}
```

Where we update the API config if it changed. The modification to the newOperConfig is done in Bootstrap function, should I be doing it somewhere else? Either way this should only happen one time if the code is executed more than once.

Reply:
ack no i won't push for a different place since its just a one time thing...lgtm